django/utils/numberformat.py

```
@@ -27,6 +27,18 @@ def format(number, decimal_sep, decimal_pos=None, grouping=0, thousand_sep='',
     # sign
     sign = ''
     if isinstance(number, Decimal):
+        # If a fixed number of decimal positions is requested, and the
+        # absolute value is smaller than the smallest representable number
+        # with that precision, force formatting as zero rather than using
+        # scientific notation.
+        if decimal_pos is not None and number != 0:
+            try:
+                if abs(number) < Decimal('1e-{}'.format(decimal_pos)):
+                    number = Decimal(0)
+            except Exception:
```

[major] This broad  silently hides errors inside the core formatting path. Internal code should fail loudly instead of swallowing unexpected states. Could we restructure the magnitude check so it doesn't rely on catching everything (for example by using /), or at least catch the specific  we expect?